tower/src/balance/p2c/mod.rs

```diff
@@ -1,4 +1,32 @@
-//! A Power-of-Two-Choices Load Balancer
+//! This module implements the "[Power of Two Random Choices]" load balancing algorithm.
+//!
+//! It is a simple but robust technique for spreading load across services with only inexact load
+//! measurements. As its name implies, whenever a request comes in, it samples two ready services
+//! at random, and issues the request to whichever service is less loaded. How loaded a service is
+//! is determiend by the return value of [`Load`](tower::load::Load).
+//!
+//! As described in the [Finagle Guide][finagle]:
+//!
+//! > The algorithm randomly picks two services from the set of ready endpoints and
+//! > selects the least loaded of the two. By repeatedly using this strategy, we can
+//! > expect a manageable upper bound on the maximum load of any server.
+//! >
+//! > The maximum load variance between any two servers is bound by `ln(ln(n))` where
+//! > `n` is the number of servers in the cluster.
+//!
+//! The balance service and layer implementations rely on you providing the underlying set of
+//! services to balance requests across. They do so using the
+//! [`Discover`](tower::discover::Discover) trait, which is essentially a `Stream` that indicates
```

Comment:
Nit/TIOLI: I might consider introducing the term "service discovery" along with the discover trait? Maybe something like 
```suggestion
//! The balance service and layer implementations rely on _service discovery_ information providing
//! the underlying set of services to balance requests across. Service discovery is represented by
//! the [`Discover`](tower::discover::Discover) trait, which is essentially a `Stream` that indicates
```